File: consumer-service/src/connections/kafka.js
import { Kafka } from "kafkajs";

export default (host) => {
  const kafkaHost = host || process.env.KAFKA_HOST || "localhost:9092";
  const kafkaClient = new Kafka({
    clientId: "producer-service",
    brokers: [kafkaHost],
  });
  console.log(`kafka host at - ${kafkaHost}`);
  return kafkaClient;
};

File: crypto-price-watcher/internal/messageapi/kafka.go
package messageapi

import (
	"github.com/ThreeDotsLabs/watermill"
	"github.com/ThreeDotsLabs/watermill-kafka/v2/pkg/kafka"
)

var (
	//TODO: get kafka host from config
	kafkaHost string = "localhost:9092"
	logger           = watermill.NewStdLogger(false, true)
)

func NewPubliser() (*kafka.Publisher, error) {
	config := kafka.PublisherConfig{
		Brokers:   []string{kafkaHost},
		Marshaler: kafka.DefaultMarshaler{},
	}
	return kafka.NewPublisher(config, logger)
}

func NewSubscriber() (*kafka.Subscriber, error) {
	config := kafka.SubscriberConfig{
		Brokers:               []string{kafkaHost},
		Unmarshaler:           kafka.DefaultMarshaler{},
		OverwriteSaramaConfig: kafka.DefaultSaramaSubscriberConfig(),
	}
	return kafka.NewSubscriber(config, logger)
}
